README.md

```diff
@@ -7,56 +7,190 @@ Lisk is a next generation crypto-currency and decentralized application platform
-  curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
-  sudo apt-get install -y nodejs
-  ```
+        ```
+        brew install git
+        ```
 
-  Locally using [nvm](https://github.com/creationix/nvm):
+### Node.js (<https://nodejs.org/>)
 
-  ```
-  curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.0/install.sh | bash
-  nvm install v6.10.1
-  ```
+- Node.js serves as the underlying engine for code execution.
+
+    Install System wide via package manager:
+
+    - Linux:
+
+        ```
+        curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
+        sudo apt-get install -y nodejs
+        ```
+
+    - Mac:
+
+        ```
+        brew install node@6
+        ```
+
+- _(Recommended)_ Install n -- Used for Node.js version management
 
-- Install PostgreSQL (version 9.6.2):
+    ```
+    npm install -g n
+    n 6.12.1
+    ```
+
+- _(Recommended)_ PM2 (<https://github.com/Unitech/pm2>) -- PM2 manages the node process for Lisk
 
   ```
-  curl -sL "https://downloads.lisk.io/scripts/setup_postgresql.Linux" | bash -
-  sudo -u postgres createuser --createdb $USER
-  createdb lisk_test
-  createdb lisk_main
-  sudo -u postgres psql -d lisk_test -c "alter user "$USER" with password 'password';"
-  sudo -u postgres psql -d lisk_main -c "alter user "$USER" with password 'password';"
+  npm install -g pm2
   ```
 
-- Bower (<http://bower.io/>) -- Bower helps to install required JavaScript dependencies.
+#### Special note about NPM 5
+
+Due to an issue with NPM 5.4.x and higher, node-sodium currently cannot be built. Therefore it is recommended to fixate the local NPM version at 5.3.x if you are running NPM 5.4.x or higher.
+
+All Systems - This may require sudo depending on your environment:
+
+```
+npm install -g npm@5.3.0
+```
+
+### PostgreSQL (version 9.6.6):
+
+   - Linux:
+
+        ```
+        curl -sL "https://downloads.lisk.io/scripts/setup_postgresql.Linux" | bash -
+        sudo -u postgres createuser --createdb $USER
+        createdb lisk_test
+        createdb lisk_main
+        sudo -u postgres psql -d lisk_test -c "alter user "$USER" with password 'password';"
+        sudo -u postgres psql -d lisk_main -c "alter user "$USER" with password 'password';"
+        ```
+
+   - Mac:
+
+        When installing a different version, replace 9.6.6 with your version in the following commands:
+
+        ```
+        brew install postgresql
+        initdb /usr/local/var/postgres -E utf8
+        mkdir -p ~/Library/LaunchAgents
+        cp /usr/local/Cellar/postgresql/9.6.6/homebrew.mxcl.postgresql.plist ~/Library/LaunchAgents/
+        lunchy start postgres
+        createdb lisk_test
+        createdb lisk_main
+        ```
+
+### Installing Redis
+
+   - Linux:
+
+        ```
+        wget http://download.redis.io/redis-stable.tar.gz
+        tar xvzf redis-stable.tar.gz
+        cd redis-stable
+        make
+        sudo make install
+        ```
+
+        Start redis-server:
+
+        ```
```

Comment:
Debian/Ubuntu: service redis start or systemctl start redis

Reply:
We are installing from source, not package manager.

Reply:
We are installing from source, not package manager.

Reply:
Done 👍